lisa/combinators/git_bisect_combinator.py

```diff
@@ -0,0 +1,191 @@
+                "Source path is not set. Please check the source clone."
+            )
+
+    @classmethod
+    def type_name(cls) -> str:
+        return constants.COMBINATOR_GITBISECT
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return GitBisectCombinatorSchema
+
+    def _next(self) -> Optional[Dict[str, Any]]:
+        _next: Optional[Dict[str, Any]] = None
+        self._process_result()
+        if not self._check_bisect_complete():
+            _next = {}
+            _next["ref"] = self._get_current_commit_hash()
+        else:
+            self._log.info("Bisect Complete")
+        self._result_notifier.result = None
+        self._iteration += 1
+        return _next
+
+    def _process_result(self) -> None:
+        if self._iteration == 0:
+            return
+        if self._result_notifier.result is not None:
+            results = self._result_notifier.result
+            if results:
+                self._bisect_good()
+            else:
+                self._bisect_bad()
+        else:
+            raise LisaException(
+                "Bisect combinator does not get result for next iteration. Please check"
+                " GitBisectResult notifier."
+            )
+
+    def _get_remote_node(self) -> Node:
+        if not self._node or not self._node.is_connected:
+            self._node = quick_connect(self.runbook.connection, "source_node")
+        return self._node
+
+    def _clone_source(self) -> None:
+        node = self._get_remote_node()
+        git = node.tools[Git]
+        self._source_path = git.clone(
+            url=self.runbook.repo, cwd=node.working_path, timeout=1200
+        )
+        node.close()
+
+    def _start_bisect(self) -> None:
+        node = self._get_remote_node()
+        git_bisect = node.tools[GitBisect]
+        git_bisect.start(cwd=self._source_path)
+        git_bisect.good(cwd=self._source_path, ref=self.runbook.good_commit)
+        git_bisect.bad(cwd=self._source_path, ref=self.runbook.bad_commit)
+        node.close()
+
+    def _bisect_bad(self) -> None:
+        node = self._get_remote_node()
+        git_bisect = node.tools[GitBisect]
+        git_bisect.bad(cwd=self._source_path)
+        node.close()
+
+    def _bisect_good(self) -> None:
+        node = self._get_remote_node()
+        git_bisect = node.tools[GitBisect]
+        git_bisect.good(cwd=self._source_path)
+        node.close()
+
+    def _check_bisect_complete(self) -> bool:
+        node = self._get_remote_node()
+        git_bisect = node.tools[GitBisect]
+        result = git_bisect.check_bisect_complete(cwd=self._source_path)
+        node.close()
+        return result
+
+    def _get_current_commit_hash(self) -> str:
+        node = self._get_remote_node()
+        git = node.tools[Git]
+        result = git.get_current_commit_hash(cwd=self._source_path)
+        node.close()
+        return result
+
+
+class GitBisectResult(notifier.Notifier):
+    @classmethod
```

Comment:
@squirrelsc I encountered `2023-10-26 07:01:34.979[904][ERROR] lisa. registered [] subclass again. It should happen in UT only. new: [<class 'lisa.runner.RunnerResult'>], exist: [<class 'lisa.combinators.git_bisect_combinator.GitBisectResult'>]` in today's testing, is it the previous PR wrong?